### PR TITLE
feat: Update SplitButton to support Heart

### PR DESCRIPTION
**Pull request by dmisdm**
This PR uplifts the SplitButton component by ensuring that tokens are switched according to the designs and execution document, for the new Heart theme.

https://www.figma.com/file/eZKEE5kXbEMY3lx84oz8iN/%E2%9D%A4%EF%B8%8F-UI-Kit-Heart?node-id=14512%3A404

https://docs.google.com/document/d/1G5iN3zZ0XHc0K5FKIFpa263nGW3l8d-jV2nWf6Gcucs/edit#

# Stories
https://dev.cultureamp.design/danno/heart-uplift/split-button/storybook/?path=/docs/splitbutton-react--anchor-link#stories

# Screenshots (if appropriate)

# Checklist

- [ ] If this contains visual changes, has it been reviewed by a designer?
- [ ] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [ ] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss [hunk 1]
```diff
@@ -101,32 +112,37 @@ $side-padding: 3/4 * $ca-grid;
...
-    $border-color: $kz-color-wisteria-200,
-    $color: $kz-color-wisteria-800,
-    $hover-background: $kz-color-stone,
-    $hover-border-color: $kz-color-wisteria-400,
-    $active-background: $kz-color-ash,
-    $active-border-color: $kz-color-wisteria-500,
-    $disabled-background: $kz-color-white,
-    $disabled-border-color: $kz-color-wisteria-200,
-    $disabled-color: $kz-color-wisteria-300
+    $background: $kz-var-color-white,
```
Comment (jasononeil) [inline draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:115]: Adding one of these to make cleanup easier:

```suggestion
    // Use of variable fallbacks to switch between Heart and Zen themes (use Zen as default fallback)
    $background: $kz-var-color-white,
```

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss [hunk 2]
```diff
@@ -101,32 +112,37 @@ $side-padding: 3/4 * $ca-grid;
...
+      var($kz-var-id-color-slate, $kz-var-color-wisteria-500),
+    $disabled-background: $kz-var-color-white,
+    $disabled-border-color:
+      var(--dt-heart-disabled-border-color, $kz-var-color-wisteria-200),
+    $disabled-color: var(--dt-heart-disabled-color, $kz-var-color-wisteria-300)
   );
 }
 
 .buttonPrimary {
   // equal to add-tint($ca-palette-seedling, 10%). Temporary until we upgrade to Zen (Wisteria)
```
Comment (jasononeil) [inline draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:131]:
```suggestion
  // #58b59a is equal to add-tint($ca-palette-seedling, 10%). Temporary until we remove Zen (Wisteria)
```

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss [hunk 3]
```diff
@@ -243,14 +259,32 @@ $side-padding: 3/4 * $ca-grid;
...
   // Making the left | right border white to separate the button with dropdown button
-  border-left-color: $kz-color-white !important; /* stylelint-disable-line declaration-no-important */
+  /* stylelint-disable declaration-no-important */
+  border-left-color: var(
+    --dt-border-left-color-heart,
+    $kz-var-color-white
+  ) !important;
 
   [dir="rtl"] & {
     // equal to add-tint($ca-palette-seedling, 10%). Temporary until we upgrade to Zen (Wisteria)
```
Comment (jasononeil) [inline draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:275]:
```suggestion
    // equal to add-tint($ca-palette-seedling, 10%). Temporary until we remove Zen
```

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss [hunk 4]
```diff
@@ -243,14 +259,32 @@ $side-padding: 3/4 * $ca-grid;
...
+    border-left-color: #58b59a !important;
+    border-left-color: var(
+      --dt-border-left-color-heart,
+      $kz-var-color-white
+    ) !important;
+  }
+
+  &:active,
+  &:hover {
+    border-left-color: transparent !important;
```
Comment (jasononeil) [inline draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:285]: Are these rules new?

**Review by jasononeil — APPROVED** [review]
This looks like a hell of a style file.

I've made minor suggestions and have one comment, but the code mostly looks good. I'm happy to approve now.

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss (hunk 4 above)
Comment (dmisdm) [reply draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:285]: Yes! something that Marc picked up that needed to change.

File: draft-packages/split-button/KaizenDraft/SplitButton/styles.scss [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
-@import "~@kaizen/design-tokens/sass/shadow";
+@import "~@kaizen/design-tokens/sass/color-vars";
+@import "~@kaizen/design-tokens/sass/border-vars";
+@import "~@kaizen/design-tokens/sass/shadow-vars";
+@import "~@kaizen/design-tokens/sass/variable-identifiers";
 @import "~@kaizen/deprecated-component-library-helpers/styles/color";
 @import "~@kaizen/deprecated-component-library-helpers/styles/type";
 @import "~@kaizen/component-library/styles/border";
 @import "~@kaizen/component-library/styles/layers";
 @import "~@kaizen/component-library/styles/animation";
```
Comment (ActuallyACat) [inline draft-packages/split-button/KaizenDraft/SplitButton/styles.scss:9]: Curious whether these dependencies are still needed? `@kaizen/component-library/*`?

**Review by ActuallyACat — APPROVED** [review]
LGTM - Cleaning up our Sass dependencies is a greater issue and can be pushed to a health task.

**Pull request merged**